invenio_xrootd/storage.py

```diff
@@ -121,20 +121,23 @@ class EOSFileStorage(XRootDFileStorage):
     clients **must** provide the file size upfront.
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, fileurl, default_booking_size=None, **kwargs):
         """Initialize storage."""
         self.bookingsize = None
-        super(EOSFileStorage, self).__init__(*args, **kwargs)
+        self.default_booking_size = default_booking_size or \
+            (current_app.config.get('MAX_CONTENT_LENGTH') if current_app
```

Comment:
Sure you can. It's just that `has_app_context` is more readable and IMHO expresses the intent better.